### PR TITLE
fix(storage): fix spill table id too strict assertion

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve https://github.com/risingwavelabs/risingwave/issues/17734

Update a previous unit test so that the corner case can be covered

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Li0k — APPROVED** [review]
LGTM

File: src/storage/src/hummock/event_handler/uploader/mod.rs [hunk 1]
```diff
@@ -965,7 +965,12 @@ impl UploaderData {
             .map(|task_id| {
                 let (sst, spill_table_ids) =
                     self.spilled_data.remove(task_id).expect("should exist");
-                assert_eq!(spill_table_ids, table_ids);
```
Comment (kwannoel) [inline src/storage/src/hummock/event_handler/uploader/mod.rs:968]: What are `spill_table_ids` and `table_ids` referring to in this context?

**Comment by kwannoel** [comment]
Triggered the job to run. May wait and see the results first.

**Review by hzxa21 — APPROVED** [review]

**Comment by kwannoel** [comment]
Oh this is to fix join amplification. Not the integration scale test. May ignore the main-cron result.

**Review by kwannoel — APPROVED** [review]

File: src/storage/src/hummock/event_handler/uploader/mod.rs (hunk 1 above)
Comment (wenym1) [reply src/storage/src/hummock/event_handler/uploader/mod.rs:968]: `spill_table_ids` is the involved table ids in a spill task. `table_ids` is the table ids to be synced.

**Comment by wenym1** [comment]
No need to cherry-pick to any release. The bug is merged recently.

**Comment by PhanSon95** [comment]
I'm staying at v1.9.1. Is this fix apply to that version already ?

**Comment by wenym1** [comment]
> I'm staying at v1.9.1. Is this fix apply to that version already ?

It seems that v1.9.1 does not even include the code that introduces the too strict assertion.

Did you fail this assertion in your environment?

**Comment by PhanSon95** [comment]
Exactly, I'm deploying risingwave to my k8s cluster in distributed mode, and today the `compactor` component crashed (image attached below)
image

**Comment by PhanSon95** [comment]
Btw, I see we already had release [v1.10.0](https://github.com/risingwavelabs/risingwave/releases/tag/v1.10.0) .
The question is, is that fix applied to this release ?

**Comment by wenym1** [comment]
> Btw, I see we already had release [v1.10.0](https://github.com/risingwavelabs/risingwave/releases/tag/v1.10.0) . The question is, is that fix applied to this release ?

I just checked the code of v1.10.0, and the failed assertion resolved in this PR had not been included in v1.10.0 either. 

In the same file there is a similar assertion `assert_eq!(epoch, epoch_data.epoch);`. Did you fail at this line instead? It will great if you can share the full panic stack trace.

**Comment by wenym1** [comment]
> Exactly, I'm deploying risingwave to my k8s cluster in distributed mode, and today the `compactor` component crashed (image attached below) image

Do you mind opening a [bug report issue](https://github.com/risingwavelabs/risingwave/issues/new?assignees=&labels=type%2Fbug&projects=&template=bug_report.yml) and fill in the related information as much as possible in the issue template so that we can fix the bug with more context? Thanks!

**Comment by PhanSon95** [comment]
@wenym1 Ofcourse, let me open bug report issue like you mention.
Here is the link of issue: https://github.com/risingwavelabs/risingwave/issues/18054
Please take a look, thanks
